src/options.rs

```diff
@@ -84,6 +84,9 @@ impl ColumnOptions {
 			log::error!(target: "parity-db", "Using `preimage` option on an ordered column is not supported");
 			return false
 		}
+		if self.ref_counted && !self.preimage {
+			log::error!(target: "parity-db", "Using `ref_counted` option without `preimage` enabled is not supported");
```

Comment:
I would return false (unless we just want the error logged).
Also I realize that the test above will block usage of ref-counted for btree, so I propose we simply remove the previous test.